docs/concepts/runtime_capabilities.md

```diff
@@ -103,21 +103,59 @@ Both the [`elastic` connector](../reference/connectors/elastic.md) and the [`htt
 
-This has the effect that we have 1 contraflow event flowing backwards for each event reaching its destination. The forward traffic volume will be mirrored by the contraflow volume, just that the contraflow itself is not leaving the system but is handled internally by [Operators] and [Connectors].
+This means that we have one contraflow event flowing backward for each event reaching its destination. The forward traffic volume will be mirrored by the contraflow volume, just that the contraflow itself is not leaving the system but is handled internally by [Operators] and [Connectors].
 
-Not all connectors support event acknowledgements, as their nature doesn't support the notion of marking parts of a data stream as successfully handled. For example [UDP](../reference/connectors/udp.md) cannot support this feature. 
+Not all connectors support event delivery guarantees, as their nature doesn't support the notion of marking parts of a data stream as successfully handled. Those connectors will acknowledge every event that reaches them. For example [UDP](../reference/connectors/udp.md) falls into this category. Other connectors only acknowledge successful transmission, such as [TCP](../reference/connectors/tcp.md) but not successful reception. Yet others only acknowledge successful reception but not necessarily successful processing of the downstream system, such as [Websockets](../reference/connectors/ws.md).
 
-Connectors like [`kafka_consumer`](../reference/connectors/kafka.md#consumer) [`wal`](../reference/connectors/wal.md) support event delivery acknowledgements and are well suited for implementing workloads that can guarantee **at-least-once* delivery.
+Connectors like [`kafka_consumer`](../reference/connectors/kafka.md#consumer) [`wal`](../reference/connectors/wal.md) support event delivery and processing acknowledgements and are well suited for implementing workloads that can guarantee *at-least-once* delivery. This guarantee can, however, only hold if all participants honor the same guarantee.
 
-For the circuit breaker mechanism a successful event acknowledgement is used a signal for the circuit breaker to close as the downstream system seems to be back to normal.
+For the circuit breaker mechanism, a successful event acknowledgment is used as a signal for the circuit breaker to close as the downstream system seems to be back to normal.
+
+Please note that contraflow events, be it circuit breaker or GD events, do not traverse over connectors. In other words, two pipelines connected by a connector will not see each other's contraflow events.
+
+### Example 1: kafka source & udp sink
+
+**Source**: The kafka source supports guaranteed delivery. A failure on a message after the latest acknowledged offset will revert to that message and replay all data since then.
```

Comment:
```suggestion
**Source**: The `kafka_consumer` connector, when used with `transactional` mode supports guaranteed delivery. A failure on a message after the latest acknowledged offset will revert to that message and replay all data since then.
```